python/cudf/cudf/_fuzz_testing/utils.py

```diff
@@ -102,6 +132,66 @@ def compare_content(a, b):
         )
 
 
+def get_avro_dtype_info(dtype):
+    if dtype in _PANDAS_TO_AVRO_SCHEMA_MAP:
+        return _PANDAS_TO_AVRO_SCHEMA_MAP[dtype]
+    else:
+        raise TypeError(
+            f"Unsupported dtype({dtype}) according to avro spec:"
+            f" https://avro.apache.org/docs/current/spec.html"
+        )
+
+
+def get_avro_schema(df):
+
+    fields = [
+        {"name": col_name, "type": get_avro_dtype_info(col_dtype)}
+        for col_name, col_dtype in df.dtypes.items()
+    ]
+    schema = {"type": "record", "name": "Root", "fields": fields}
+    return schema
+
+
+def convert_nulls_to_none(records, df):
+    columns_with_nulls = {col for col in df.columns if df[col].isnull().any()}
+    scalar_columns_convert = [
+        col
+        for col in df.columns
+        if df[col].dtype in pandas_dtypes_to_cudf_dtypes
+        or pd.api.types.is_datetime64_dtype(df[col].dtype)
+        or pd.api.types.is_timedelta64_dtype(df[col].dtype)
+    ]
+
```

Comment:
It seems like we might be able to accomplish the goal of this process without having to drop to a list of dicts first and then loop over them. If feasible I'd suggest exploring if the columns can be cast to `object` in pandas and then modified using `.apply` with a lambda.